zerver/lib/bugdown/fenced_code.py

```diff
@@ -107,6 +108,24 @@
 CODE_WRAP = '<pre><code%s>%s\n</code></pre>'
 LANG_TAG = ' class="%s"'
 
+def validate_curl_content(lines: List[str]) -> None:
+    error_msg = """
+Missing required -X argument in curl command:
+
+{command}
+""".strip()
+
+    for line in lines:
+        regex = r"curl \-X \"?(GET|DELETE|PATCH|POST)\"?"
```

Comment:
I generally recommend (1) using a different quote (e.g. `'`) rather than escaping `\` and (2) using `[-]` rather than `\-` (it's more readable).